os/arch/arm/src/imxrt/imxrt_wdog_lowerhalf.c

```diff
@@ -0,0 +1,254 @@
+static int imxrt_wdog_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status);
+static int imxrt_wdog_settimeout(FAR struct watchdog_lowerhalf_s *lower, uint32_t timeout);
+static xcpt_t imxrt_wdog_capture(FAR struct watchdog_lowerhalf_s *lower, xcpt_t handler);
+static int imxrt_wdog_ioctl(FAR struct watchdog_lowerhalf_s *lower, int cmd, unsigned long arg);
+
+const struct watchdog_ops_s g_wdgops = {
+	.start = imxrt_wdog_start,
+	.stop = imxrt_wdog_stop,
+	.keepalive = imxrt_wdog_keepalive,
+	.getstatus = imxrt_wdog_getstatus,
+	.settimeout = imxrt_wdog_settimeout,
+	.capture = imxrt_wdog_capture,
+	.ioctl = imxrt_wdog_ioctl,
+};
+
+static int imxrt_wdog_start(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_init(priv->base, &priv->config);
+	return OK;
+}
+
+static int imxrt_wdog_stop(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_deinit(priv->base);
+
+	return OK;
+}
+
+static int imxrt_wdog_keepalive(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_refresh(priv->base);
+
+	return OK;
+}
+
+static int imxrt_wdog_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	status->flags = imxrt_wdog_getstatusflags(priv->base);
+	status->timeout = priv->config.timeoutValue;
+	/* TO BE UPDATED */
+	status->timeleft = 0;
+
+	return OK;
+}
+
+static int imxrt_wdog_settimeout(FAR struct watchdog_lowerhalf_s *lower, uint32_t timeout)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	/* Set to new timeout. */
+	priv->config.timeoutValue = (uint16_t)timeout;
+
+	/* Restart the watchdog with new configuration. */
+	imxrt_wdog_init(priv->base, &priv->config);
+
+	return OK;
+}
+
+static xcpt_t imxrt_wdog_capture(FAR struct watchdog_lowerhalf_s *lower, xcpt_t handler)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	xcpt_t oldhandler;
+
+	DEBUGASSERT(priv);
+
+	oldhandler = priv->handler;
+
+	/* Set the new handler. */
+	priv->handler = handler;
+
+	/* Restart the watchdog with new configuration. */
+	imxrt_wdog_init(priv->base, &priv->config);
+
+	return oldhandler;
+}
+
+static int imxrt_wdog_ioctl(FAR struct watchdog_lowerhalf_s *lower, int cmd, unsigned long arg)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	switch (cmd) {
+	case WDIOC_PAUSE: {
+		/* Set WDOG_WCR_WDW to 1 for suspending WDOG timer operation. */
+		priv->config.workMode.enableWait = 1;
```

Comment:
1. It set/reset enableWait only in ioctl.
    Let's assume that below case.
    start -> ioctl(pause) -> start
    It will work properly?
2. enableStop also there, when it set? you didn't set it even if stop called.
3. we didn't support wait mode for watchdog before, is it new requirement?
4. If there's new requirement, you have to thing about getstatus.
    I think below is needed.
```
   if (lower->getstatus) |= wait) {
      ioctl(resume(;
   }
```
5. If you think it wait logic is generic operation for all other devices & user should know it then you can expand flag of watchdog_status_s.

Reply:
1. Yes. it is okay. because `imxrt_wdog_init` initializes the config and starts the timer again.
2. `imxrt_wdog_deinit` disable the wdog.
3,4. yes, this is a requirement. I will change the code soon.
5. I will consider it. Thanks.